azurelinuxagent/ga/exthandlers.py

```diff
@@ -300,13 +300,13 @@ def run(self):
         try:
             extensions_goal_state = self.protocol.get_extensions_goal_state()
 
-            # self.ext_handlers needs to be initialized first, since status reporting depends on it
+            # self.ext_handlers and etag need to be initialized first, since status reporting depends on them
             self.ext_handlers = extensions_goal_state.extensions
+            etag = self.protocol.client.get_goal_state().incarnation
```

Comment:
despite the name of this local, this needs to be the incarnation number for status reporting to work - I introduced a regression a couple of PRs ago and was using the 'id' property; now i am fixing this to use the incarnation explicitly

there are a lot renames/refactoring needed in this module, i am planning on doing those gradually